packages/core/helper-plugin/src/hooks/useQueryParams.ts

```diff
@@ -34,7 +36,7 @@ const useQueryParams = (initialParams) => {
     [push, query]
   );
 
-  return [{ query, rawQuery: search }, setQuery];
+  return [{ query, rawQuery: search }, setQuery] as const;
```

Comment:
nice use ✅  

For future if you weren't aware (and neither is better imo) you could also do a named tuple as the return type:

```ts
const useQueryParams = <T extends Record<string, JSON>>(initialParams?: T): [query: X, setQuery: Y] => {
```